classes/view/members.php

```diff
@@ -0,0 +1,143 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Members view class deals with generating the HTM for the members table. It
+ * can generate members table for either tutors or students in a course by
+ * calling the "build_members_view" method. Defaults to rendering the students
+ * members table if no display role is given.
+ */
+class members_view {
+    public $course;
+    public $coursemodule;
+    public $turnitintooltwoassignment;
+    public $turnitintooltwoview;
+
+    public function __construct($course=null, $coursemodule=null, $turnitintooltwoview=null, $turnitintooltwoassignment=null) {
+        $this->course = $course;
+        $this->coursemodule = $coursemodule;
+        $this->turnitintooltwoview = $turnitintooltwoview;
+        $this->turnitintooltwoassignment = $turnitintooltwoassignment;
+    }
+
+    /**
+     * Method that generates the members view HTML. Depending on the displayrole
+     * passed will generate the HTML for the users with that role.
+     * @return string Members HTML for a role
+     */
+    public function build_members_view($displayrole = "students") {
+        $output  = "";
+        $istutor = $this->is_tutor();
+        $memberrole = $this->get_role_for_display_role($displayrole);
+
+        if (!$istutor) {
+            turnitintooltwo_print_error('permissiondeniederror', 'turnitintooltwo');
+            exit();
+        }
+
+        // wrapper element for strong CSS selectors
+        if ($displayrole == "tutors") {
+            $output .= html_writer::start_tag("div", array("class" => "members members-instructors"));
+        } else {
+            $output .= html_writer::start_tag("div", array("class" => "members members-students"));
+        }
```

Comment:
This could be better if the class name was a variable, then we can get rid of some of the duplicate lines above.